--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,9 @@
 
 add_library(${PROJECT_NAME} INTERFACE)
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    target_compile_options(${PROJECT_NAME} INTERFACE /wd5245)
+endif()
 
 if(NOT CMAKE_VERSION VERSION_LESS 3.8)
     target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
@@ -68,7 +71,7 @@
 ################################################################################
 
 if(${DOCTEST_WITH_MAIN_IN_STATIC_LIB})
-    add_library(${PROJECT_NAME}_with_main STATIC EXCLUDE_FROM_ALL ${doctest_parts_folder}/doctest.cpp)
+    add_library(${PROJECT_NAME}_with_main STATIC ${doctest_parts_folder}/doctest.cpp)
     target_compile_definitions(${PROJECT_NAME}_with_main PRIVATE
         DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN)
     set_target_properties(${PROJECT_NAME}_with_main PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON)
@@ -130,7 +133,7 @@
 
 if(NOT ${DOCTEST_NO_INSTALL})
     install(
-        TARGETS ${PROJECT_NAME}
+        TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_with_main
         EXPORT "${targets_export_name}"
         INCLUDES DESTINATION "${include_install_dir}"
     )
